src/app/util/ember-compatibility-functions.cpp

```diff
@@ -94,5 +157,141 @@ bool ServerClusterCommandExists(chip::ClusterId aClusterId, chip::CommandId aCom
+        break;
+    }
+    case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
+    {
+        uint64_t uint64_data;
+        memcpy(&uint64_data, data, sizeof(uint64_data));
+        ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), uint64_data));
+        break;
+    }
+    case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
+    {
+        int8_t int8_data;
+        memcpy(&int8_data, data, sizeof(int8_data));
+        ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), int8_data));
+        break;
+    }
+    case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
+    {
+        int16_t int16_data;
+        memcpy(&int16_data, data, sizeof(int16_data));
+        ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), int16_data));
+        break;
+    }
+    case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
+    {
+        int32_t int32_data;
+        memcpy(&int32_data, data, sizeof(int32_data));
+        ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), int32_data));
+        break;
+    }
+    case ZCL_INT64S_ATTRIBUTE_TYPE: // Signed 64-bit integer
+    {
+        int64_t int64_data;
+        memcpy(&int64_data, data, sizeof(int64_data));
+        ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), int64_data));
+        break;
+    }
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE: // Octet string
+    {
+        uint8_t * actualData = data + 1;
+        uint8_t dataLength   = data[0];
+        ReturnErrorOnFailure(
+            apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), chip::ByteSpan(actualData, dataLength)));
+        break;
+    }
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE: {
+        uint8_t * actualData = data + 2; // The pascal string contains 2 bytes length
+        uint16_t dataLength;
+        memcpy(&dataLength, data, sizeof(dataLength));
+        ReturnErrorOnFailure(
+            apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), chip::ByteSpan(actualData, dataLength)));
+        break;
+    }
+    case ZCL_ARRAY_ATTRIBUTE_TYPE: {
+        // TODO: Add support to structures
+        TLV::TLVType containerType;
+        ReturnErrorOnFailure(
+            apWriter->StartContainer(TLV::ContextTag(AttributeDataElement::kCsTag_Data), TLV::kTLVType_List, containerType));
+        // TODO: Encode data in TLV, now raw buffers
+        ReturnErrorOnFailure(apWriter->PutBytes(
+            TLV::AnonymousTag, data,
+            emberAfAttributeValueSize(aClusterInfo.mClusterId, aClusterInfo.mFieldId, metadata->attributeType, data)));
+        ReturnErrorOnFailure(apWriter->EndContainer(containerType));
+        break;
+    }
+    default:
+        ChipLogError(DataManagement, "Attribute type %d not handled", static_cast<int>(dataType));
```

Comment:
Please make the output the attribute code in hex. I'm pretty sure nobody is used to look at the attribute code in decimal.